File: wfm-develop mobile 07.05.2026/wfm/mobile/ios/WFMApp/Core/Navigation/MainTab.swift
import SwiftUI

/// Табы главного экрана
enum MainTab: Int, CaseIterable {
    case home = 0
    case tasks = 1
    case control = 2
    case settings = 3

    /// Заголовок таба
    var title: String {
        switch self {
        case .home: return "Главная"
        case .tasks: return "Задачи"
        case .control: return "Контроль"
        case .settings: return "Профиль"
        }
    }

    /// Иконка таба (из Assets)
    var icon: String {
        switch self {
        case .home: return "ic-tab-home"
        case .tasks: return "ic-tab-tasks"
        case .control: return "ic-tab-control"
        case .settings: return "ic-tab-profile"
        }
    }

    /// Стартовый таб
    static let startTab: MainTab = .home

    /// Табы для работника (без control)
    static let workerTabs: [MainTab] = [.home, .tasks, .settings]

    /// Табы для менеджера (с control)
    static let managerTabs: [MainTab] = [.home, .tasks, .control, .settings]
}

File: wfm-develop mobile 07.05.2026/wfm/mobile/android/app/src/main/java/com/beyondviolet/wfm/navigation/MainTab.kt
package com.beyondviolet.wfm.navigation

import androidx.annotation.DrawableRes
import com.beyondviolet.wfm.R

/**
 * Описание табов главного экрана
 *
 * @param route Уникальный маршрут таба
 * @param title Заголовок для отображения
 * @param icon Ресурс иконки таба
 */
sealed class MainTab(
    val route: String,
    val title: String,
    @DrawableRes val icon: Int
) {
    data object Home : MainTab(
        route = "tab_home",
        title = "Главная",
        icon = R.drawable.ic_tab_home
    )

    data object Tasks : MainTab(
        route = "tab_tasks",
        title = "Задачи",
        icon = R.drawable.ic_tab_tasks
    )

    data object Control : MainTab(
        route = "tab_control",
        title = "Контроль",
        icon = R.drawable.ic_tab_control
    )

    data object Settings : MainTab(
        route = "tab_settings",
        title = "Профиль",
        icon = R.drawable.ic_tab_profile
    )

    companion object {
        /** Табы для работника (без Control) */
        val workerEntries = listOf(Home, Tasks, Settings)

        /** Табы для менеджера (с Control) */
        val managerEntries = listOf(Home, Tasks, Control, Settings)

        /** Стартовый таб */
        val startTab = Home
    }
}
